Lib/test/test_asyncio/test_timeouts.py

```diff
@@ -298,6 +299,18 @@ async def test_nested_timeout_in_finally(self):
         self.assertIs(e2.__context__, e22)
 
     async def test_timeout_after_cancellation(self):
+        asyncio.current_task().cancel()
+        with self.assertRaises(asyncio.CancelledError):
+            async with asyncio.timeout(0.1):
+                await asyncio.sleep(1)
+
+    async def test_timeout_immediately_after_cancellation(self):
+        asyncio.current_task().cancel()
+        with self.assertRaises(asyncio.CancelledError):
+            async with asyncio.timeout(0.0):
+                await asyncio.sleep(1)
```

Comment:
this added test fails on all python versions and the proposed fix makes it pass without breaking any other tests